instrumentation/sofa-rpc/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/sofa/rpc/internal/SofaRpcClientNetworkAttributesGetter.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.sofa.rpc.internal;
+
+import com.alipay.sofa.rpc.client.ProviderInfo;
+import com.alipay.sofa.rpc.context.RpcInternalContext;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import io.opentelemetry.instrumentation.api.semconv.network.NetworkAttributesGetter;
+import io.opentelemetry.instrumentation.api.semconv.network.ServerAttributesGetter;
+import io.opentelemetry.instrumentation.sofa.rpc.SofaRpcRequest;
+import java.net.InetSocketAddress;
+import javax.annotation.Nullable;
+
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
+public final class SofaRpcClientNetworkAttributesGetter
+    implements ServerAttributesGetter<SofaRpcRequest>,
+        NetworkAttributesGetter<SofaRpcRequest, SofaResponse> {
+
+  @Nullable
+  @Override
+  public String getServerAddress(SofaRpcRequest request) {
+    ProviderInfo providerInfo = getProviderInfo();
+    if (providerInfo != null) {
+      return providerInfo.getHost();
+    }
+
+    // Fallback to remote address
+    InetSocketAddress remoteAddress = request.remoteAddress();
```

Comment:
`server.address `should reflect the server where the request was sent to. If we can't get correct value, we'd return null instead of wrong value.